src/operations/blas2/xpmv.hpp

```diff
@@ -176,12 +190,13 @@ Spmv<lhs_t, matrix_t, vector_t, local_range_x, local_range_y, is_upper>::eval(
 
   // VERTICAL
 
-  {
+  if (is_symmetric || is_transposed) {
```

Comment:
I think you can use `if constexpr (is_symmetric || is_transposed)` to move this check to compile time.

Reply:
Thanks for your review, @aacostadiaz.

I understand your concerns, but the fact that `is_simmetric` and `is_transposed` are two template parameters is enough for allowing clang to evaluate these if statements on compile time.